### PR TITLE
Integrate unit-tests for function-list

### PR DESCRIPTION
**Pull request by donho**
Implement #8757, close #3393

`unitTestLauncher.ps1` should be launched by Appveyor, if this script returns -1 then build should fail, otherwise it will return 0 so build continues.

### CODE REVIEW

File: PowerEditor/Test/FunctionList/bash/unitTest.expected.result [hunk 1]
```diff
@@ -1 +1 @@
-{"leaves":["setenv","action_needed","canonicalize_path","make_module","db2_name","versioned_copy"],"root":"unitTest"}
\ No newline at end of file
+{"leaves":["action_needed","canonicalize_path","make_module","db2_name","versioned_copy"],"root":"unitTest"}
```
Comment (donho) [inline PowerEditor/Test/FunctionList/bash/unitTest.expected.result:1]: "setenv" should be present IMO, so there has been already a regression.

**Comment by donho** [comment]
@GitMensch @chcg Here the working unit-tests for function-list.
Any idea to integrate it into appveyor.yml ?

**Comment by chcg** [comment]
@donho It did a first version, see https://github.com/chcg/notepad-plus-plus/tree/issue_8762 with result https://ci.appveyor.com/project/chcg/notepad-plus-plus/builds/34984122. The problem seems to come from scintilla SciLexer.dll which is not build with boost regex support by appveyor. Probably the CI build could be modified to use the prebuild boost libraries from the appveyor environment, see https://www.appveyor.com/docs/windows-images-software/#boost

I will give that a try.

PS:
Additionally there seems to be a problem with the powershell script, if it runs on the #net core version (https://github.com/PowerShell/PowerShell/releases/tag/v7.0.3)

**Comment by chcg** [comment]
Boost is just available as source code, but not with binary libs for regex. So probably the easiest check would be against some prebuild version like the last release e.g or at least the SciLexer.dll from such a build.

**Pull request closed without merging**